docs/framework-overview.md

```diff
@@ -0,0 +1,398 @@
+    subgraph source_phase [A. Arrangement]
+        direction TB
+        parse[A1. Parse EasyConfig]
+        fetch[A2. Fetch sources]
+        check[A3. Check readiness]
+        unpack[A4. Unpack sources]
+        patch[A5. Apply patches]
+        parse --> fetch
+        fetch --> check
+        check --> unpack
+        unpack --> patch
+    end
+    subgraph build_phase [B. Build and Install]
+        direction TB
+        prepare[B1. Prepare build]
+        config[B2. Configure build]
+        build[B3. Compile/build]
+        test[B4. Test]
+        install[B5. Intallation]
+        prepare --> config
+        config --> build
+        build --> test
+        test --> install
+    end
+    subgraph ext_phase [Extensions]
+        ext_install[Extension Install]
+    end
+    subgraph clean_phase [C. Cleaning]
+        direction TB
+        postinstall[C1. Post-Install]
+        sanity[C2. Sanity check]
+        clean[C3. Clean-up]
+        module[C4. Env. Module]
+        perms[C5. Permissions]
+        package[C6. Package]
+        testcases[C7. Test Cases]
+        postinstall --> sanity
+        sanity --> clean
+        clean --> module
+        module --> perms
+        perms --> package
+        package --> testcases
+    end
+    source_phase --> build_phase
+    build_phase --> ext_phase
+    ext_phase --> clean_phase
+```
+
+## The EasyBlock class
+
+The base EasyBlock class in `easybuild.framework.easyblock` is a skeleton class
+that cannot carry out any installation on its own, but that lays the ground for
+child classes to define custom installation procedures. This does not mean that
+the EasyBlock class is an empty shell though, it is actually one of the biggest
+classes in EasyBuild with 4000+ lines of code (v4.7.x). It provides methods for
+generic actions that are useful in most installations independently of the
+installation method.
+
+Most of the steps in the initial [*Arrangement phase*][build_install_steps] are
+already covered to some extent in the EasyBlock class, such as:
+
+* A2. Fetch sources: `EasyBlock.fetch_sources`, `EasyBlock.fetch_patches`,
+  `EasyBlock.obtain_file`
+
+* A3. Check readiness: `EasyBlock.make_buildir`, `EasyBlock.reset_env`
+
+* A4. Unpack sources: `EasyBlock.checksum_step`, `EasyBlock.extract_step`
+
+Several steps in the last [*Cleaning phase*][build_install_steps] are already
+covered by the base EasyBlock as well:
+
+* C1. Post-Install: `EasyBlock.run_post_install_commands`,
+  `EasyBlock.apply_post_install_patches`, `EasyBlock.fix_shebang`
+
+* C2. Sanity check: `EasyBlock.sanity_check_step`,
+  `EasyBlock.sanity_check_load_module`,
+  `EasyBlock.sanity_check_linked_shared_libs`
+
+* C3. Clean-up: `EasyBlock.cleanup_step`
+
+* C4. Env. Module: `EasyBlock.make_module_step`
+
+* C5. Permissions: `EasyBlock.permissions_step`
+
+Therefore, custom easyblocks (child classes of `EasyBlock`) only need to define
+the steps of the [*Build and Install phase*][build_install_steps], the core of
```

Comment:
```suggestion
the steps of the [*Build and Install phase*][step_wise_installation_procedure], the core of

Reply:
This statement isn't fully correct: only the `configure_step`, `build_step`, and `install_step` methods of `EasyBlock` need to be defined.

`prepare_step` usually doesn't need to be redefined at all, and redefining `test_step` is entirely optional...